lib/ClangImporter/ImportName.cpp

```diff
@@ -1405,7 +1405,11 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
   case clang::DeclarationName::CXXUsingDirective:
   case clang::DeclarationName::CXXDeductionGuideName:
     // Handling these is part of C++ interoperability.
-    llvm_unreachable("unhandled C++ interoperability");
+    if (getContext().LangOpts.EnableCXXInterop) {
+      return ImportedName();
+    } else {
+      llvm_unreachable("unhandled C++ interoperability");
```

Comment:
I think it's fine to do this unconditionally now, instead of checking for `EnableCXXInterop`. `llvm_unreachable` turns into an optimization hint in release builds anyway.